src/mmda/predictors/heuristic_predictors/figure_table_predictors.py

```diff
@@ -110,58 +351,58 @@ def _predict(doc: Document, caption_type: str = 'Figure') -> List[SpanGroup]:
-
         predictions = []
-        for page in range(len(tqdm(doc.images))):
-            if merged_boxes_caption_dict.get(page) and merged_boxes_fig_dict.get(page):
+        predictions_captions = []
+        predictions_relations = []
+        for page in range(len(tqdm(self.doc.pages))):
+            if merged_boxes_caption_dict.get(page) and merged_boxes_fig_tab_dict.get(page):
                 cost_matrix = np.zeros(
-                    (len(merged_boxes_fig_dict[page]),
+                    (len(merged_boxes_fig_tab_dict[page]),
                      len(merged_boxes_caption_dict[page])))
-                for j, fig_box in enumerate(merged_boxes_fig_dict[page]):
-                    for i, caption_box in enumerate(merged_boxes_caption_dict[page]):
+                for j, fig_box in enumerate(merged_boxes_fig_tab_dict[page]):
+                    for i, span_group in enumerate(merged_boxes_caption_dict[page]):
+                        caption_box = span_group.box
                         assert hasattr(fig_box, 'box')
-                        assert hasattr(caption_box, 'box')
                         distance = FigureTablePredictions._get_object_caption_distance(
-                            fig_box.box, caption_box.box)
+                            fig_box.box, caption_box)
 
                         cost_matrix[j][i] = distance
                         if caption_type == 'Figure':
                             cost_matrix[j][i] = distance if distance > 0 else 900
 
                 row_ind, col_ind = linear_sum_assignment(cost_matrix)
                 for row, col in zip(row_ind, col_ind):
-                    predictions.append(SpanGroup(spans=[Span(
+                    span_group = SpanGroup(spans=[Span(
                         start=merged_boxes_caption_dict[page][col].start,
                         end=merged_boxes_caption_dict[page][col].end,
                         box=merged_boxes_caption_dict[page][col].box)],
-                        box_group=BoxGroup(boxes=[merged_boxes_fig_dict[page][row].box]),
+                        box_group=BoxGroup(boxes=[merged_boxes_fig_tab_dict[page][row].box]),
                         metadata=Metadata(type=caption_type)
-                    ))
-        return predictions
+                    )
+                    span_group.text = self.doc.symbols[span_group.start: span_group.end]
+                    if span_group.text.lower().startswith(caption_type.lower()[:3]):
+                        span_group.id = hash(json.dumps(span_group.to_json()))
+                        box_group = span_group.box_group
+                        box_group.id = hash(json.dumps(box_group.to_json()))
+                        predictions.append(box_group)
+                        predictions_captions.append(span_group)
+                        predictions_relations.append(Relation(from_id=span_group.id, to_id=box_group.id))
```

Comment:
I think encoding the relationship in either direction is fine, since it's 1-1. However, we should be consistent.

The names for your fields are:
"figure_to_figure_captions" and "table_to_table_captions"

but in this code you are making the caption the "from" entity, and the figure/table the "to" entity. Please either reverse your `from_id` and `to_id` assignment here, or rename your fields to:
"figure_caption_to_figures" and "table_captions_to_tables"